internal/controller/controllers/infraenv_controller.go

```diff
@@ -199,6 +200,7 @@ func (r *InfraEnvReconciler) processNMStateConfig(ctx context.Context, log logru
 // ensureISO generates ISO for the cluster if needed and will update the condition Reason and Message accordingly.
 // It returns a result that includes ISODownloadURL.
 func (r *InfraEnvReconciler) ensureISO(ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv) (ctrl.Result, error) {
+	infraEnv.Status.AgentLabelSelector = metav1.LabelSelector{MatchLabels: map[string]string{aiv1beta1.InfraEnvNameLabel: infraEnv.Name}}
```

Comment:
just making sure that it will not cause a reconcile loop right?

Reply:
No. Only if there is a change.